src/tools/run-make-support/src/lib.rs

```diff
@@ -169,6 +169,38 @@ pub fn cwd() -> PathBuf {
     env::current_dir().unwrap()
 }
 
+// FIXME(Oneirical): This will no longer be required after compiletest receives the ability
+// to manipulate read-only files. See https://github.com/rust-lang/rust/issues/126334
+/// Ensure that the path P is read-only while the test runs, and restore original permissions
+/// at the end so compiletest can clean up.
+#[track_caller]
+pub fn test_while_readonly<P: AsRef<Path>, F: FnOnce() + std::panic::UnwindSafe>(
+    path: P,
+    closure: F,
+) {
+    let path = path.as_ref();
+    if is_windows() && path.is_dir() {
+        eprintln!("This helper function cannot be used on Windows to make directories readonly.");
+        eprintln!(
+            "See the official documentation:
+            https://doc.rust-lang.org/std/fs/struct.Permissions.html#method.set_readonly"
+        );
+        eprintln!("You will need to add //@ ignore-windows to your rmake.rs test.");
+        panic!("Useless `test_while_readonly` detected.");
```

Comment:
Suggestion: I don't think this is a recommendation we can make, e.g. imagine if test writer accidentally passed a wrong path, that doesn't mean they should `//@ ignore-windows` neccessarily. The "useless" message is also kinda misleading. Instead, I think we should just report exactly what we have detected and no more: detected `test_while_readonly` usage while on Windows.